util/set/smallset_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package set
+
+import (
+	"fmt"
+	"iter"
+	"maps"
+	"slices"
+	"testing"
+)
+
+func TestSmallSet(t *testing.T) {
+	t.Parallel()
+
+	type op struct {
+		add bool
+		v   int
+	}
+	ops := iter.Seq[op](func(yield func(op) bool) {
+		for _, add := range []bool{false, true} {
+			for v := 1; v <= 3; v++ {
```

Comment:
This test doesn't exercise the rules about zero values, which are different from `Set` so difficult to fit in here. Maybe a small separate test?

```go
func TestSmallSet_zero(t *testing.T) {
	s := &SmallSet[int]{}
	checkInvariants := func() {
		if got, want := s.Len(), 0; got != want {
			t.Errorf("wrong len: got %v, want %v", got, want)
		}
		if s.Contains(0) {
			t.Error("set contains zero value")
		}
		for v := range s.Values() {
			t.Errorf("unexpected iterated value %v", v)
		}
	}
	checkInvariants()
	s.Add(0)
	checkInvariants()
	s.Delete(0)
	checkInvariants()
	s.Delete(0)
	checkInvariants()
}
```

Reply:
With @dsnet's comment below, this is now unnecessary, as zero is now handled.